### PR TITLE
[AIRFLOW-5282] Add default timeout on kubeclient & catch HTTPError

### PR DESCRIPTION
**Pull request by rolanddb**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues:
  - https://issues.apache.org/jira/browse/AIRFLOW-5282

### Description

This PR:
* Adds a default _request_timeout in airflow.cfg, so that default installations will not hang indefinitely
* Catches the errors in the scheduler, and retries scheduling the task, in a similar fashion to how ApiExcptions are caught. (https://github.com/apache/airflow/blob/46885ccdbc8618f295b8f986da95c1abbb85bb02/airflow/executors/kubernetes_executor.py#L800-L803)
* Fixes a bug in the parsing of airflow.cfg. The notes state that a json dict is expected. This is parsed with `json.loads`, which returns a dict. But in the code, .iteritems() is called on this dict, which cannot work. .items() should be used instead.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -776,11 +776,13 @@ tolerations =
...
 # See:
 #   https://raw.githubusercontent.com/kubernetes-client/python/master/kubernetes/client/apis/core_v1_api.py
-kube_client_request_args =
+# Note that if no _request_timeout is specified, the kubernetes client will wait indefinitely for kubernetes
+# api responses, which will cause the scheduler to hang. The timeout is specified as [connect timeout, read timeout]
 
 # Worker pods security context options
 # See:
 #   https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+kube_client_request_args = {"_request_timeout" : [60,60] }
```
Comment (ashb) [inline airflow/config_templates/default_airflow.cfg]: This should be above the `# Worker pods security context options` line.

File: airflow/contrib/executors/kubernetes_executor.py [hunk 2]
```diff
@@ -797,6 +799,10 @@ def sync(self):
                     self.log.warning('ApiException when attempting to run task, re-queueing. '
                                      'Message: %s' % json.loads(e.body)['message'])
                     self.task_queue.put(task)
+                except HTTPError as e:
+                    self.log.warning('HTTPError when attempting to run task, re-queueing. '
+                                     'Exception: %s' + str(e))
```
Comment (ashb) [inline airflow/contrib/executors/kubernetes_executor.py]:
```suggestion
                                     'Exception: %s', str(e))
```

**Review by ashb — CHANGES_REQUESTED** [review]

**Review by ashb — APPROVED** [review]

**Comment by rolanddb** [comment]
@andy-g14 I added the timeout configuration from your unit test as a default. This works at runtime, but tests are complaining about a KeyError.

https://github.com/apache/airflow/commit/f70a6aaab2424413dee03bd89bd7118f4a67f427#diff-2fd5967071561042847ef71c5f8b7e8dR216

Did you ever see this work? I tried various changes, e.g. `kube_client_request_args = "hello world"` runs fine, but it seems that adding json there breaks things..

File: airflow/config_templates/default_airflow.cfg [hunk 3]
```diff
@@ -776,7 +776,9 @@ tolerations =
 # List of supported params in **kwargs are similar for all core_v1_apis, hence a single config variable for all apis
 # See:
 #   https://raw.githubusercontent.com/kubernetes-client/python/master/kubernetes/client/apis/core_v1_api.py
-kube_client_request_args =
+# Note that if no _request_timeout is specified, the kubernetes client will wait indefinitely for kubernetes
+# api responses, which will cause the scheduler to hang. The timeout is specified as [connect timeout, read timeout]
+kube_client_request_args = {"_request_timeout" : [60,60] }
```
Comment (ashb) [inline airflow/config_templates/default_airflow.cfg]:
```suggestion
kube_client_request_args = {{"_request_timeout" : [60,60] }}
```

The problem is this file isn't a plain config file, it's actually a giant python format string.

Comment (ashb) [reply airflow/config_templates/default_airflow.cfg]: Setting this in the `~/airflow/airflow.cfg` wouldn't have the same problem, it's just this default config template that is the problem.

**Comment by andy-g14** [comment]
@rolanddb , you will need to provide the value as json string, i.e enclosed in single quotes. it works for me this way
 kube_client_request_args = '{"_request_timeout" : [10,100] }'

**Comment by ashb** [comment]
The problem in the other PR seemed to be around putting a default value in our config template (which is actually a giant python format string so `{}` -> `{{}}`)

**Comment by rolanddb** [comment]
> The problem in the other PR seemed to be around putting a default value in our config template (which is actually a giant python format string so `{}` -> `{{}}`)

Ah, I was very confused why it didn't work!

**Comment by ashb** [comment]
Nice!

Note that this new default config won't change the value for any existing installs, so some people might have to update their configs to get this new change.

**Pull request merged**